From 9b39a985bcd6cd34f0820f3680f145d46c0e56bd Mon Sep 17 00:00:00 2001
From: Titouan Christophe <titouan.christophe@railnova.eu>
Date: Sun, 8 Dec 2019 01:55:59 +0100
Subject: [PATCH] Allow avro C to be built on musl based systems.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The type `ssize_t` is defined in sys/types.h, and nowhere else
in the musl standard C library, so it should be included for the
compilation to succeed.

This fixes several errors like:

    In file included from src/generic.c:29:0:
    src/generic.c: In function ‘avro_generic_value_new’:
    src/avro_generic_internal.h:63:39:
        error: ‘ssize_t’ undeclared (first use in this function);
               did you mean ‘size_t’?

Upstream-Status: Backport
Signed-off-by: Titouan Christophe <titouan.christophe@railnova.eu>
---
 lang/c/src/avro_generic_internal.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lang/c/src/avro_generic_internal.h b/lang/c/src/avro_generic_internal.h
index 709403c03..9843ed652 100644
--- a/lang/c/src/avro_generic_internal.h
+++ b/lang/c/src/avro_generic_internal.h
@@ -24,6 +24,8 @@ extern "C" {
 #define CLOSE_EXTERN
 #endif
 
+#include <sys/types.h>
+
 #include "avro/generic.h"
 #include "avro/schema.h"
 #include "avro/value.h"
-- 
2.24.1

